starlette/requests.py

```diff
@@ -118,7 +118,7 @@ def url_for(self, name: str, **path_params: typing.Any) -> str:
 class Request(HTTPConnection):
     def __init__(self, scope: Scope, receive: Receive = empty_receive):
         super().__init__(scope)
-        assert scope["type"] == "http"
+        assert scope["type"] in ("http", "websocket")
```

Comment:
We don't want this. Instead we should change `AuthenticationMiddleware.asgi` to use `conn = HTTPConnection(scope)` and `auth_result = await self.backend.authenticate(conn)`.

Signatures in `starlette/authentication.py` should also change to use the `HTTPConnection`base class instead of `Request`.

Reply:
That way the authentication will be suitable for either HTTP requests or WebSocket connections.

Reply:
Okay I see your point about using the parent `HTTPConnection` class and I've made that change.

What are your thoughts about the default error handler? It calls `PlainTextResponse` and that seems to be a http specific response (it inherits from `Response` and has message type `http.response.start`).

Is there an error response that can work for both http and websocket? I reckon for a websocket connection, the socket should be closed at this point.